apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -126,11 +126,22 @@ pub mod oracle_process {
         // awaiting a certain number of confirmations
         let mut latest_block;
         let mut pending: Vec<PendingEvent> = Vec::new();
+        const SLEEP_DUR: std::time::Duration =
+            std::time::Duration::from_secs(1);
         loop {
+            tokio::time::sleep(SLEEP_DUR).await;
             // update the latest block height
             latest_block = loop {
-                if let Ok(height) = oracle.eth_block_number().await {
-                    break height;
+                match oracle.eth_block_number().await {
+                    Ok(height) => break height,
+                    Err(error) => {
+                        tracing::warn!(
+                            ?error,
+                            "Couldn't get the latest Ethereum block height, \
+                             will keep trying"
+                        );
+                        tokio::time::sleep(SLEEP_DUR).await;
```

Comment:
this just means the task kept running even after Tokio has begun shutting down, and it panicked because there was no longer a runtime to schedule it. to avoid this panic, I guess we would need an `abort_recv` here similar to the one in the Tendermint process handler.

Reply:
I tried some things out and couldn't get this to work, my tokio-fu isn't good enough at the moment. The oracle only receives a `Sender<()>` during construction, which it can use to indicate it can shut down. We can set up another `(abort_send, abort_recv)` channel in `run_aux`, pass the `abort_recv` to the oracle like you suggest, then `run_aux` should send on `abort_send` when the shutdown happens due to some other component. The `run_aux` code is getting really hairy now.

Reply:
I will split the oracle rate limiting out of this PR